pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -553,6 +553,22 @@ func (c *Reconciler) createTaskRun(rprt *resources.ResolvedPipelineRunTask, pr *
 			PodTemplate:        pr.Spec.PodTemplate,
 		}}
 
+	for _, w := range rprt.PipelineTask.Workspaces {
+		taskWorkspaceName := w.Name
+		pipelineWorkspaceName := w.Workspace
+		var binding *v1alpha1.WorkspaceBinding
+		for i := range pr.Spec.Workspaces {
+			if pr.Spec.Workspaces[i].Name == pipelineWorkspaceName {
+				binding = pr.Spec.Workspaces[i].DeepCopy()
+				binding.Name = taskWorkspaceName
+			}
```

Comment:
If I'm not mistaken, there isn't any validation that a pipeline run's workspaces are uniquely named like there is on the pipeline. This could lead to a weird foot-gun where the last similarly named workspace wins out. Unlikely but easily avoidable with a map of the pipeline run's workspace instead of iterating over both. 

Just a suggestion. I think this would work almost all of the time as is.

Reply:
I've added some validation around the case you've mentioned and updated this code to use a map.  I've also improved on the error message that gets returned here to be more correct.